app/controllers/users_controller.rb

```diff
@@ -1,4 +1,7 @@
 class UsersController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_user, only: %i[show edit update destroy]
+
   def index
```

Comment:
- [ ] I could not log in as I was rendered with an error.  the error suggests that the User object associated with the form is not properly initialized, as indicated by the values nil for all the attributes of the User object. 
- To fix this issue, you need to ensure that you have a valid User object available in the view. 


### View:
![image](https://github.com/lugard1/blog/assets/50990925/16d79f7a-912e-4579-aee5-ef8b7919b927)

Reply:
Fix the UsersController issue.